File: src/easy/valid_palindrome.rs
#![allow(dead_code)]
pub fn is_palindrome(s: String) -> bool {}

#[cfg(test)]
mod tests {
    use super::*;

    #[test]
    fn test_is_not_palindrome() {
        assert_eq!(is_palindrome("race a car".to_string()), false);
    }

    #[test]
    fn test_is_palindrome() {
        assert_eq!(is_palindrome(" ".to_string()), true);

        assert_eq!(is_palindrome("".to_string()), true);

        assert_eq!(
            is_palindrome("A man, a plan, a canal: Panama".to_string()),
            true
        );
    }
}

File: alternative/easy/valid_palindrome.py
def isPalindrome(s: str) -> bool:
    new_s = ""
    for c in s:
        if c.isalnum():
            new_s += c.lower()

    return new_s == new_s[::-1]

assert isPalindrome("A man, a plan, a canal: Panama") == True
assert isPalindrome("A b") == False
assert isPalindrome("A") == True
assert isPalindrome("") == True
assert isPalindrome(" ") == True
assert isPalindrome("a") == True